operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java

```diff
@@ -357,18 +354,23 @@ public R updateStatus(R resource) {
       return (R) hasMetadataOperation.replaceStatus(resource);
     }
 
-    public R patchStatus(R resource) {
+    public R patchStatus(R resource, R originalResource) {
       log.trace("Updating status for resource: {}", resource);
       String resourceVersion = resource.getMetadata().getResourceVersion();
-      try {
-        // don't do optimistic locking on patch
-        resource.getMetadata().setResourceVersion(null);
+      // don't do optimistic locking on patch
+      originalResource.getMetadata().setResourceVersion(null);
+      resource.getMetadata().setResourceVersion(null);
+      try (var bis = new ByteArrayInputStream(
```

Comment:
Maybe put a todo/comment that this is currently needed until 6.0?

Reply:
yep, will add  comment. Actually rather create an issue (todo -s should be just temporal until PR is merged in my definition :) )